src/oidc_provider_mock/__main__.py

```diff
@@ -108,17 +114,36 @@ def run(
             if not isinstance(claims_dict, dict):
                 raise click.ClickException("--user-claims must be a JSON object")
 
-            sub = claims_dict.get("sub")
-            if not sub or not isinstance(sub, str):
-                raise click.ClickException(
-                    '--user-claims must include a "sub" property'
-                )
-
-            claims = {k: v for k, v in claims_dict.items() if k != "sub"}
-            user_claims_list.append(User(sub=sub, claims=claims))
+            user_claims_list.append(_decode_claims_dict(claims_dict))
         except json.JSONDecodeError as e:
             raise click.ClickException(f"Invalid JSON in --user-claims: {e}") from e
 
+    if user_claims_file:
+        if not os.path.isfile(user_claims_file):
+            raise click.ClickException("--user-claims-file must point to a valid file.")
+
+        try:
+            with open(user_claims_file) as json_file:
+                user_claims_file_list: list[dict[str, object] | None] | None = (
+                    json.load(json_file)
```

Comment:
How about accepting YAML? It is a superset of JSON so JSON will be accepted as well. I prefer writing configuration in YAML because it easier